### PR TITLE
Require x.dims=label.dims in huber_loss

### PR DESCRIPTION
**Pull request by Aurelius84**
+ Align to `tf.losses.huber_loss` that requires x.dims=label.dims
+ Remove last dimentsion forced to be 1.

### CODE REVIEW

**Review by sneaxiy — APPROVED** [review]
LGTM.

**Review by heavengate — APPROVED** [review]
